packages/gradle/src/utils/exec-gradle.ts

```diff
@@ -53,3 +68,35 @@ export function execGradleAsync(
     });
   });
 }
+
+/**
+ * This function recursively finds the nearest gradlew file in the workspace
+ * @param filePath the path to start searching for gradlew file
+ * @returns the relative path of the gradlew file to workspace root, throws an error if gradlew file is not found
+ * It will return gradlew.bat file on windows and gradlew file on other platforms
+ */
+export function findGraldewFile(
+  filePath: string,
+  wr: string = workspaceRoot
+): string {
+  const fileDirectory = dirname(filePath);
+  const gradlewPath = join(dirname(filePath), 'gradlew');
+  const gradlewBatPath = join(dirname(filePath), 'gradlew.bat');
+  if (process.platform.startsWith('win')) {
+    if (existsSync(join(wr, gradlewBatPath))) {
+      return gradlewBatPath;
+    }
+  } else {
+    if (existsSync(join(wr, gradlewPath))) {
+      return gradlewPath;
+    }
+  }
+
+  if (!fileDirectory || fileDirectory === '.') {
```

Comment:
We can move this check up... and use

`if (path === dirname(path)) {` then we throw this error